pkgs/applications/misc/appmenu-gtk-module/default.nix

```
@@ -0,0 +1,39 @@
+{ stdenv, fetchFromGitLab, fetchurl, meson, cmake, pkg-config, systemd, gtk-doc, ninja, glib, gtk2, gtk3 }:
+let
+xinitfile = fetchurl {
+      url = "https://raw.githubusercontent.com/archlinux/svntogit-community/packages/appmenu-gtk-module/trunk/80-appmenu-gtk-module";
+      sha256 = "1g4zjqnr89achb8k97j26f0dl8wy7afm4dbdmmq60mdqlx76q02c";
+};
+in
+stdenv.mkDerivation {
+  pname = "appmenu-gtk-module";
+  version = "0.7.3.2";
+  src = fetchFromGitLab {
```

To preserve the `gtk-doc`-generated documentation:

```suggestion

  outputs = [ "out" "devdoc" ];

  src = fetchFromGitLab {
```

---

I tried adding this but ending up disabling the line because it resulted in build error:

> building '/nix/store/jsbpc067ahrylvy2s6ymqn8gxwn70gjy-source.drv'...
> 
> trying https://gitlab.com/api/v4/projects/rilian-la-te%2Fvala-panel-appmenu/repository/archive.tar.gz?sha=0.7.3.2
>   % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
>                                  Dload  Upload   Total   Spent    Left  Speed
>   0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0
> curl: (22) The requested URL returned error: 404
> error: cannot download source from any mirror
> builder for '/nix/store/jsbpc067ahrylvy2s6ymqn8gxwn70gjy-source.drv' failed with exit code 1
> cannot build derivation '/nix/store/2zcvfl8dmw7pvviylc81a5098vbr72pz-appmenu-gtk-module-0.7.3.2.drv': 1 dependencies couldn't be built
> cannot build derivation '/nix/store/kb4ks0xmvdx8g7vf14ryqzbv44fpy5qr-system-path.drv': 1 dependencies couldn't be built
> cannot build derivation '/nix/store/iqkirandk9244dgx0my2bdyhw6xvmdqz-nixos-system-louis-t25-20.09.git.966e6d44f6fM.drv': 1 dependencies couldn't be built
> error: build of '/nix/store/iqkirandk9244dgx0my2bdyhw6xvmdqz-nixos-system-louis-t25-20.09.git.966e6d44f6fM.drv' failed

---

That looks like you did not change the owner, which is `vala-panel-project` on GitLab, not `rilian-la-te`.

---

Fixed now. However now I get this error:

> find: '/nix/store/p6d5cb5cdcs1znxami9zqzaj4rkyqhyr-appmenu-gtk-module-0.7.3.2-devdoc': No such file or directory
> strip is /nix/store/1ar9gcmvikdk1wij0cw4fnwvl5i07ag1-binutils-2.31.1/bin/strip
> builder for '/nix/store/9z4akjfa1kdir943h9g10d5jbcivifjb-appmenu-gtk-module-0.7.3.2.drv' failed to produce output path '/nix/store/p6d5cb5cdcs1znxami9zqzaj4rkyqhyr-appmenu-gtk-module-0.7.3.2-devdoc'

---

Looks like the docs are no longer installed: https://gitlab.com/vala-panel-project/vala-panel-appmenu/-/commit/bb3e3b3bc4875b120e64a62e374b02ddf23771e3

Try applying the following patch using `fetchpatch`:

https://gitlab.com/vala-panel-project/vala-panel-appmenu/commit/7bc0ae409f8878dfe81b913564e20296b9593407.patch

---

I had to make a local patch as the working directory while building differs from the patch due to "sourceRoot".
Now I get this error:

> Building documentation for appmenu-gtk-parser
> ERROR: Error in gtkdoc helper script:
> 
> ERROR: ['/nix/store/7fwa4h2h7f52a1s6mrxpkl7dd00idiss-gtk-doc-1.32/bin/gtkdoc-mkhtml', '--path=/build/source/subprojects/appmenu-gtk-module/docs:/build/source/subprojects/appmenu-gtk-module/build/docs', 'appmenu-gtk-parser', '..//build/source/subprojects/appmenu-gtk-module/build/docs/appmenu-gtk-parser-docs.sgml'] failed with status 5
> I/O error : Attempt to load network entity http://docbook.sourceforge.net/release/xsl/current/html/chunk.xsl
> warning: failed to load external entity "http://docbook.sourceforge.net/release/xsl/current/html/chunk.xsl"
> compilation error: file /nix/store/7fwa4h2h7f52a1s6mrxpkl7dd00idiss-gtk-doc-1.32/share/gtk-doc/data/gtk-doc.xsl line 11 element import
> xsl:import : unable to load http://docbook.sourceforge.net/release/xsl/current/html/chunk.xsl

I find this strange as I'm able to download the file via Firefox to it seems to be up.

---

I would just do `cd subprojects/appmenu-gtk-module` in `preConfigure`. Unfortunately, the `sourceRoot` is too finicky. That will allow us to use regulat patches.

Yes, the error is because Nix builds packages in a sandbox without access to internet by default. Add `docbook-xsl-nons` to `nativeBuildInputs` to make the needed file available.

---

Done both now. Seems to work as desired.